packages/connections/src/utils/validation-errors.ts

```diff
@@ -0,0 +1,148 @@
+import { ConnectionInfo } from 'mongodb-data-service';
+import ConnectionString from 'mongodb-connection-string-url';
+import {
+  FormValidationError,
+  getConnectionString,
+  isSecure,
+} from './validation';
+
+function validateConnectionInfoErrors(
+  connectionInfo: ConnectionInfo
+): FormValidationError[] {
+  const connectionString = getConnectionString(connectionInfo);
+
+  return [
+    ...validateAuthMechanism(connectionString),
+    ...validateUsernameAndPassword(connectionString),
+    ...validateLocalhostAndTls(connectionString),
+    ...validateSSHTunnel(connectionInfo),
+    ...validateKerberosPrincipal(connectionString),
+    ...validateLDAP(connectionString),
+    ...validateX509(connectionString),
+  ];
+}
+
+function validateAuthMechanism(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  return errors;
+}
+
+function validateX509(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  if (connectionString.searchParams.get('authMechanism') === 'MONGODB-X509') {
+    if (!isSecure(connectionString)) {
+      errors.push({
+        message: 'TLS must be enabled in order to use x509 authentication.',
+      });
+    }
+    if (!connectionString.searchParams.has('tlsCertificateFile')) {
+      errors.push({
+        message: 'A Client Certificate is required with x509 authentication.',
+      });
+    }
+  }
+
+  return errors;
+}
+
+function validateLDAP(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  if (connectionString.searchParams.get('authMechanism') === 'PLAIN') {
+    if (!connectionString.username) {
+      errors.push({
+        field: 'ldapUsername',
+        message: 'LDAP username is required.',
+      });
+    }
+
+    if (!connectionString.password) {
+      errors.push({
+        message: 'Username and password are required.',
+      });
+    }
+  }
+  return errors;
+}
+function validateKerberosPrincipal(
+  connectionString: ConnectionString
+): FormValidationError[] {
+  const errors: FormValidationError[] = [];
+  if (
+    connectionString.searchParams.get('authMechanism') === 'GSSAPI' &&
+    !connectionString.username
+  ) {
+    errors.push({
+      field: 'kerberosPrincipal',
+      message: 'Principal name is required with Kerberos.',
+    });
+  }
+  return errors;
+}
+function validateSSHTunnel(
+  connectionInfo: ConnectionInfo
+): FormValidationError[] {
+  const connectionOptions = connectionInfo.connectionOptions;
+  if (!connectionOptions.sshTunnel) {
+    return [];
+  }
+  const errors: FormValidationError[] = [];
+  if (!connectionOptions.sshTunnel.host) {
```

Comment:
**suggestion**: can we also validate if the host is valid or not? (@Anemy implemented [this](https://github.com/mongodb-js/compass/blob/main/packages/connect-form/src/utils/check-for-invalid-character-in-host.ts))